src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs

```diff
@@ -113,27 +116,32 @@ internal string FormatScriptCommand(string scriptCommandline, string scriptExten
             return command;
         }
 
-        internal string ReplaceScriptVariables(string command)
+        internal string ReplaceScriptVariables(string scriptCommandline)
         {
-            foreach (var scriptVariableEntry in ScriptVariableToMSBuildMap)
+            Func<string, string> scriptVariableReplacementDelegate = key =>
             {
-                var scriptVariableName = scriptVariableEntry.Key;
-                var msbuildMapping = scriptVariableEntry.Value;
-
-                if (command.Contains($"%{scriptVariableName}%"))
+                if (ScriptVariableToMSBuildMap.ContainsKey(key))
                 {
-                    if (msbuildMapping == null)
+                    if (ScriptVariableToMSBuildMap[key] == null)
                     {
                         MigrationErrorCodes.MIGRATE1016(
-                                $"{scriptVariableName} is currently an unsupported script variable for project migration")
+                                $"{key} is currently an unsupported script variable for project migration")
                             .Throw();
                     }
 
-                    command = command.Replace($"%{scriptVariableName}%", msbuildMapping);
+                    return ScriptVariableToMSBuildMap[key];
                 }
-            }
+                return $"$({key})";
+            };
 
-            return command;
+            var scriptArguments = CommandGrammar.Process(
+                scriptCommandline,
+                scriptVariableReplacementDelegate,
+                preserveSurroundingQuotes: true);
+
+            scriptArguments = scriptArguments.Where(argument => !string.IsNullOrEmpty(argument)).ToArray();
```

Comment:
Pulled this straight from the implementation in ScriptExecutor, I'd like to keep it as near to that as possible so you can eyeball the comparison if need be.